### DISCUSSION TITLE
How to set the maximum line of Theia editor code

### DISCUSSION THREAD

**Discussion by zx616456607** [eclipse-theia/theia#13443] [Q&A] [answered]
I am using EditorManager, how to set the maximum number of lines of editor code, such as 9999, beyond the maximum number of lines cannot be written

**Comment by msujew** [comment, marked as answer]
Hey @zx616456607,

The monaco editor doesn't expose any way to limit the number of lines users can write. You might want to raise this as an issue on the monaco/vscode issue board.

Aside from that, you might be able to get away to listen to the `onDocumentContentChanged` event of each editor and simply call `undo` on any change that increases the lines of code after 9999 LoC have been reached.
